File: android/tutorials/tutorial1/src/test/java/com/uber/rib/root/task_act/TaskActInteractorTest.java
package com.uber.rib.root.task_act;

import com.uber.rib.core.RibTestBasePlaceholder;
import com.uber.rib.core.InteractorHelper;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

public class TaskActInteractorTest extends RibTestBasePlaceholder {

  @Mock TaskActInteractor.TaskActPresenter presenter;
  @Mock TaskActRouter router;

  private TaskActInteractor interactor;

  @Before
  public void setup() {
    MockitoAnnotations.initMocks(this);

    interactor = TestTaskActInteractor.create(presenter);
  }

  /**
   * TODO: Delete this example and add real tests.
   */
  @Test
  public void anExampleTest_withSomeConditions_shouldPass() {
    // Use InteractorHelper to drive your interactor's lifecycle.
    InteractorHelper.attach(interactor, presenter, router, null);
    InteractorHelper.detach(interactor);

    throw new RuntimeException("Remove this test and add real tests.");
  }

}

File: android/tutorials/tutorial1/src/test/java/com/uber/rib/root/task_act/TaskActInteractorTest.kt
package com.uber.rib.root.task_act

import com.uber.rib.core.RibTestBasePlaceholder
import com.uber.rib.core.InteractorHelper

import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.MockitoAnnotations

class TaskActInteractorTest : RibTestBasePlaceholder() {

  @Mock internal lateinit var presenter: TaskActInteractor.TaskActPresenter
  @Mock internal lateinit var router: TaskActRouter

  private var interactor: TaskActInteractor? = null

  @Before
  fun setup() {
    MockitoAnnotations.initMocks(this)

    interactor = TestTaskActInteractor.create(presenter)
  }

  /**
   * TODO: Delete this example and add real tests.
   */
  @Test
  fun anExampleTest_withSomeConditions_shouldPass() {
    // Use InteractorHelper to drive your interactor's lifecycle.
    InteractorHelper.attach<TaskActInteractor.TaskActPresenter, TaskActRouter>(interactor!!, presenter, router, null)
    InteractorHelper.detach(interactor!!)

    throw RuntimeException("Remove this test and add real tests.")
  }
}